ENH: Add Makefile for local development testing of build

matthewfeickert opened the pull request:
Resolves #7 

Add a very simple `Makefile` for local development testing to build the image. This allows for using

```
make
```

to build the image and

```
make run
```

to drop into an interactive session in the image.

This should **wait** for PR #6 to get merged so that this can be rebased and add in build args.

matthewfeickert committed:
ENH: Add Makefile for local development

matthewfeickert committed:
MNT: Bind mount $PWD in image in 'run' target

This allows for the user's current working directory to get bind mounted
into the container at /mpl_source and also to have the working directory
get set to /mpl_source so they are ready to go

QuLogic commented on Makefile:
```diff
@@ -0,0 +1,14 @@
+
+run:
+	docker run --rm -it \
+	-v $(shell pwd):/mpl_source \
```
But you should be mounting the Matplotlib source, not this directory.

matthewfeickert replied:
No, not for testing the build of _this image_.

QuLogic replied:
There's nothing in this repo to test with.

matthewfeickert replied:
The `Makefile` is local to this project. What I mean is that if you're working on refactoring the image and you want to build things locally and then drop into it to look around and play then this `Make` target is helpful. For use in actually doing things in `matplotlib` that's a different story. But what @tacaswell [is demoing in PR #5](https://github.com/matplotlib/mpl-docker/pull/5#issuecomment-657091073) wouldn't ever be using this `Makefile`. That would need some shell alias or something.

Sorry, I may have a different idea in my head about what the intended use case of this specific repository is. I thought that this was for just developing the Docker image for a canonical matplotlib build/dev environment. Am I on the same train of thought as you, or did I get confused?

matthewfeickert replied:
oh sorry! I think that my choice of `mpl_source` is confusing here(?). I know that this doesn't contain the matplotlib source code, but just thought that as this would be used for matplotlib source that I'd hint at what this image would be used for. If this is bad I'll change it.

matthewfeickert committed:
MNT: Add build args to Makefile

Additionally, as a style change use BASE_IMAGE instead of BUILDER_IMAGE.
This is due to Matthew pasting in some code from a multistage Docker
build project, but BUILDER makes little sense here, so use a better
naming.

matthewfeickert commented:
As @tacaswell has pointed out, graphical forwarding is needed so I'll need to add in additional bind mounting to the example `run` `Make` target. That will need

```
-v /tmp/.X11-unix/:/tmp/.X11-unix/
```

but environment `DISPLAY` also need to get properly exported with `-e`. I've forgotten how to properly do this, so I'll need to go look at things.
